src/user-has-access-react.component.tsx

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { getCurrentUser } from "@openmrs/esm-api";
+
+export default function UserHasAccessReact(props) {
+  const [user, setUser] = React.useState(null);
+
+  function hasAccess(requiredPrivilege) {
+    //@ts-ignore
+    return user.privileges.find(p => requiredPrivilege === p.display);
+  }
+
+  React.useEffect(() => {
+    const subscription = getCurrentUser({ includeAuthStatus: false }).subscribe(
+      x => {
```

Comment:
Errors should be handled, but are not. It's one of the infrastructure things i have in my list to make sure we handle errors correctly.

In this case, you're not providing a second argument to `subscribe()` which is the error handler. For now, we can probably just do `err => {throw err}`, but in the future we'll have fancier error handling.

Reply:
Removed for now as per recs below